WaveGen/Arduino/basicStepperDriver/basicStepperDriver.ino

```diff
@@ -0,0 +1,183 @@
+#include <Arduino.h>
+#include "BasicStepperDriver.h"
+
+// motor : https://www.omc-stepperonline.com/de/p-series-ip67-wasserdicht-nema-23-schrittmotor-5-0a-1-8nm-254-95oz-in-23ip67-20
+// stepper controller: https://www.omc-stepperonline.com/de/digitaler-schrittmotortreiber-1-0-4-2a-20-50vdc-fuer-nema-17-23-24-schrittmotor-dm542t
+
+// the motor has a 1.8 deg step so 200 steps
+#define MOTOR_STEPS 200
+
+// set a safty RPM
+#define SAFTY_RPM 600
+
+// step resolution on stepper controller set to 800 step/rev
+#define MICROSTEPS 4
+
+// All the wires needed for full functionality
+#define DIR 4
+#define STEP 3
+#define ENABLE 2
+
+//Uncomment line to use enable/disable functionality
+//#define SLEEP 13
+
+// set all value to default 
+float rpmVal = 60.0;  
+float timeVal = 0.0;  
+int angleVal = 0; 
+int start_delay = 0;
+
+unsigned long start_time = 0;
+
+// start bsic driver
+BasicStepperDriver stepper(MOTOR_STEPS, DIR, STEP, ENABLE);
+
+void setup() {
+
+   // enable serial
+   Serial.begin(9600);
+   while (!Serial) {;}  // Wait for serial port to connect
+   Serial.println("Found serial connection of wavegen");
+
+   stepper.begin(rpmVal, MICROSTEPS);
+   stepper.setEnableActiveState(LOW);
+
+   stepper.disable();
+  
+}
+
+void loop(){
+
+   // parse the message from python
+   if (Serial.available() > 0) {
+
+      // read in line 
+      String input = Serial.readStringUntil('\n');
+      input.trim();
+
+      // Split command and value
+      int spaceIndex = input.indexOf(' ');
+      String command = input;
+      int value = 0;
+
+      // Split string into tokens
+      const int maxTokens = 10;
+      String tokens[maxTokens];
+      int tokenCount = 0;
+      
+      // sort the messages
+      while (input.length() > 0 && tokenCount < maxTokens) {
+         int spaceIndex = input.indexOf(' ');
+         if (spaceIndex == -1) {
+
+            tokens[tokenCount++] = input;
+            break;
+         
+         } else {
+         
+            tokens[tokenCount++] = input.substring(0, spaceIndex);
+            input = input.substring(spaceIndex + 1);
+            input.trim();
+         
+         }
+      }
+
+      // Parse name/value pairs
+      for (int i = 0; i < tokenCount - 1; i += 2) {
+      String name = tokens[i];
+      float value = tokens[i + 1].toFloat();
+
+         if (name == "a") {
```

Comment:
Kann das arduino-SDK strings mit `==` vergleichen? In C geht das nicht, da braucht man typischerweise `strcmp(name, "a") == 0`